bson/slice_codec.go

```diff
@@ -99,6 +137,12 @@ func (sc *sliceCodec) DecodeValue(dc DecodeContext, vr ValueReader, val reflect.
 		return ValueDecoderError{Name: "SliceDecodeValue", Kinds: []reflect.Kind{reflect.Slice}, Received: val}
 	}
 
+	if vr.Type() == TypeBinary {
```

Comment:
The logic defined in this block needs to be an extension of the TypeBinary switch case.